src/mesh/tests/BoundingBoxTest.cpp

```diff
@@ -13,6 +13,29 @@ using namespace precice::mesh;
 BOOST_AUTO_TEST_SUITE(MeshTests)
 BOOST_AUTO_TEST_SUITE(BoundingBoxTests)
 
+BOOST_AUTO_TEST_CASE(Constructor)
+{
+  PRECICE_TEST(1_rank);
+  { // 3D
+    Eigen::VectorXd boundMin(3);
+    boundMin << 1.0, 2.0, 3.0;
+    Eigen::VectorXd boundMax(3);
+    boundMax << 4.0, 5.0, 6.0;
+    BoundingBox bb(boundMin, boundMax);
+    BoundingBox compareData({1.0, 4.0, 2.0, 5.0, 3.0, 6.0});
```

Comment:
So, here you are testing the new constructor compared to the old constructor, right?
If we remove the old constructor, what happens here?

Reply:
The purpose of the test is to make sure that the new BB constructor returns the correct values. But we get what you mean and we think we should remove the old constructor and just check if the new constructor works as expected.

```cpp
BOOST_AUTO_TEST_CASE(Constructor)
{
  PRECICE_TEST(1_rank);
  { // 3D
    Eigen::VectorXd boundMin(3);
    boundMin << 1.0, 2.0, 3.0;
    Eigen::VectorXd boundMax(3);
    boundMax << 4.0, 5.0, 6.0;
    BoundingBox bb(boundMin, boundMax);

    BOOST_TEST(bb.getDimension() == 3);
    BOOST_TEST(bb.minCorner() == boundMin);
    BOOST_TEST(bb.maxCorner() == boundMax);
  }
  { // 2D
    Eigen::VectorXd boundMin(2);
    boundMin << 1.0, 2.0;
    Eigen::VectorXd boundMax(2);
    boundMax << 4.0, 5.0;
    BoundingBox bb(boundMin, boundMax);
    
    BOOST_TEST(bb.getDimension() == 2);
    BOOST_TEST(bb.minCorner() == boundMin);
    BOOST_TEST(bb.maxCorner() == boundMax);
  }
} // Constructor
```